src/lib/containers/entity/annotations/AnnotationEditorUtils.ts

```diff
@@ -42,56 +46,60 @@ export function getFriendlyPropertyName(error: AjvError) {
+      const errorGroup = grouped[property]
 
-    // First, see if it is an anyOf error
-    const hasAnyOfError = errorGroup.some(
-      e => e.message === 'should match some schema in anyOf',
-    )
+      // First, see if it is an anyOf error
+      const hasAnyOfError = errorGroup.some(
+        e => e.message === 'should match some schema in anyOf',
+      )
 
-    // We determine if it's an anyOf *enum* error if all error messages in the property match one of these three messages:
-    const isEnumError =
-      hasAnyOfError &&
-      errorGroup.every(error => {
-        if (error.message === 'should be string') {
-          return true
-        } else if (error.message === 'should be equal to constant') {
-          return true
-        } else if (error.message === 'should match some schema in anyOf') {
-          return true
-        } else {
-          return false
-        }
-      })
+      // We determine if it's an anyOf *enum* error if all error messages in the property match one of these three messages:
+      const isEnumError =
+        hasAnyOfError &&
+        errorGroup.every(error => {
+          if (error.message === 'should be string') {
+            return true
+          } else if (error.message === 'should be equal to constant') {
+            return true
+          } else if (error.message === 'should match some schema in anyOf') {
+            return true
+          } else {
+            return false
+          }
+        })
 
-    // If it's an anyOf enum error, we just modify the first message and drop the rest
-    if (isEnumError && errorGroup.length > 0) {
-      errorGroup[0].message = 'should be equal to one of the allowed values'
-      grouped[property] = [errorGroup[0]]
-    }
-  })
+      // If it's an anyOf enum error, we just modify the first message and drop the rest
+      if (isEnumError && errorGroup.length > 0) {
+        errorGroup[0].message = 'should be equal to one of the allowed values'
+        grouped[property] = [errorGroup[0]]
+      }
+    })
 
-  // Ungroup the errors after potentially modifying them
-  errors = flatMap(grouped)
+    // Ungroup the errors after potentially modifying them
+    errors = flatMap(grouped)
 
-  // Custom error message if the custom annotation key collides with an internal entity property
-  errors = errors.map(error => {
-    const propertyName = getFriendlyPropertyName(error)
-    if (entityJsonKeys.includes(propertyName)) {
-      error.message = `"${propertyName}" is a reserved internal key and cannot be used.`
+    // Custom error message if the custom annotation key collides with an internal entity property
+    if (concreteType) {
+      errors = errors.map(error => {
+        const propertyName = getFriendlyPropertyName(error)
+        if (entityJsonKeys[concreteType].includes(propertyName)) {
+          error.message = `"${propertyName}" is a reserved internal key and cannot be used.`
+        }
+        return error
+      })
     }
```

Comment:
Only non-whitespace change in the inner function - wrap this transformation operation in `if (concreteType)` and change `entityJsonKeys` to `entityJsonKeys[concreteType]`